sycl/test-e2e/NumComputeUnits/num_compute_units.cpp

```diff
@@ -0,0 +1,21 @@
+// This test checks whether the number of compute units in the device descriptor
```

Comment:
Considering that we now support the extension, we should also define and test presence of the corresponding [feature test macro](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_num_compute_units.asciidoc#feature-test-macro)